README.md

```diff
@@ -18,6 +18,10 @@ With this in mind, there are many tenants or principles that we follow in order
 
 - We support microcontrollers with no operating system, microcontrollers with a real-time operating system (like [Azure RTOS](http://rtos.com)), Linux, and Windows. Customers can implement their own "platform layer" to use our SDK on devices we don’t support out-of-the-box. The platform layer requires minimal functionality such as a clock, a mutex, sleep, and an HTTP stack. We provide some platform layers, and more will be added over time.
 
+## Documentation

```

Comment:
nit: Consider moving this down to right above the `Contributing` section (and right after `SDK Architecture`).

The reason is because I think the GitHub repo and the getting started is more relevant/useful to customers than knowing where the docs link is.

Here's why:
1) On GitHub, at the top of the repo, we already broadcast the docs link
![image](https://user-images.githubusercontent.com/6527137/84550893-b9418a80-acc0-11ea-9162-71c1e881c63c.png)

2) If already looking at the doxygen docs, via https://azure.github.io/azure-sdk-for-c, mentioning this link again, is redundant since the customer is already on that page.

Up to you though, just a nit.

Reply:
The new proper link for Azure rtos is this: https://azure.microsoft.com/en-us/services/rtos/

Reply:
@ahsonkhan I might be adding my bias here but it took me an embarrassing amount of time to find that link at the top; I never thought to look there. For almost every repo I use, I usually (for better or worse) scroll right to the Readme and try to find a documentation section. I was hoping to make that front and center and make it really easy to find. That way, people can understand what they are "getting started" with before actually "getting started".